src/main/scala/is/hail/utils/IntervalTree.scala

```diff
@@ -131,23 +165,24 @@ object IntervalTree {
 case class IntervalTreeNode[U](i: Interval,
   left: Option[IntervalTreeNode[U]],
   right: Option[IntervalTreeNode[U]],
-  maximum: Any, value: U) extends Traversable[(Interval, U)] {
+  minimum: Any, maximum: Any, value: U) extends Traversable[(Interval, U)] {
 
   def contains(pord: ExtendedOrdering, position: Any): Boolean = {
-    pord.lteq(position, maximum) &&
+    pord.lteq(position, maximum) && pord.gteq(position, minimum) &&
       (left.exists(_.contains(pord, position)) ||
         (pord.gteq(position, i.start) &&
           (i.contains(pord, position) ||
             right.exists(_.contains(pord, position)))))
   }
 
   def overlaps(pord: ExtendedOrdering, interval: Interval): Boolean = {
-    pord.lteq(interval.start, maximum) && (left.exists(_.overlaps(pord, interval))) ||
+    pord.lteq(interval.start, maximum) && pord.gteq(interval.end, minimum) &&
+      (left.exists(_.overlaps(pord, interval))) ||
       i.overlaps(pord, interval) || (right.exists(_.overlaps(pord, interval)))
   }
 
   def query(pord: ExtendedOrdering, b: mutable.Builder[Interval, _], position: Any) {
-    if (pord.lteq(position, maximum)) {
+    if (pord.lteq(position, maximum) && pord.gteq(position, minimum)) {
```

Comment:
Why is comparing to the minimum necessary here? Is it just to make it faster?

Reply:
yeah, just to make it faster I think.